File: 0832-flipping-an-image/0832-flipping-an-image.c
int** flipAndInvertImage(int** image, int imageSize, int* imageColSize, int* returnSize, int** returnColumnSizes) {
    int** result = (int**)malloc(sizeof(int*) * imageSize);
    *returnSize = imageSize;
    *returnColumnSizes = (int*)malloc(sizeof(int) * imageSize);

    for (int i = 0; i < imageSize; i++) {
        result[i] = (int*)malloc(sizeof(int) * (*imageColSize));
        (*returnColumnSizes)[i] = *imageColSize;

        for (int j = 0; j < *imageColSize; j++) {
            result[i][j] = 1 - image[i][(*imageColSize - 1) - j];
        }
    }

    return result;
}
void printMatrix(int** matrix, int rows, int cols) {
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

File: 0832-flipping-an-image/0832-flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        int m=image.length,n=image[0].length;
        int[][] arr=new int[m][n];
        for(int i=0;i<m;i++){
            int x=0,y=n-1;
            while(x<=y){
                arr[i][x]=image[i][y];
                arr[i][y]=image[i][x];
                x++; y--;
            }
        }
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(arr[i][j]==0)arr[i][j]=1;
                else arr[i][j]=0;
            }
        }
        return arr;
    }
}